STAR Fusion Tool
================

-   [Description](#description)
    -   [CGC Location](#cgc-location)
-   [Preview Command](#preview-command)
-   [Common issues](#common-issues)
-   [Contatiner](#contatiner)
-   [I/O](#io)
    -   [Inputs](#inputs)
    -   [Outputs](#outputs)
    -   [Arguments](#arguments)

Description
===========

STAR-Fusion, next to being a component of STAR 2.5, is also a component of the Trinity Cancer Transcriptome Analysis Toolkit (CTAT). STAR-Fusion uses the STAR aligner to identify candidate fusion transcripts supported by Illumina reads. STAR-Fusion further processes the output generated by the STAR aligner to map junction reads and spanning reads to a reference annotation set. A reference genome and corresponding protein-coding gene annotation set, including blast-matching gene pairs must be provided to STAR-Fusion.

### CGC Location

<https://igor.sbgenomics.com/u/dave/cgrrna/apps/#dave/cgrrna/star-fusion-2-5-1-b/>

Preview Command
===============

    tar -xvzf /path/to/GRCh37_gencode_v19_CTAT_lib_July272016.tar.gz  -C . && /opt/STAR-Fusion_v0.7.0_FULL/STAR-Fusion --chimeric_junction /path/to/sample1.Chimeric.out.junction --tmpdir /tmp --verbose_level 2 --output_dir fusion_output  && tar -vcf sample1.fusion_output.tar.gz fusion_output && mv fusion_output/star-fusion.fusion_candidates.preliminary fusion_output/sample1.star-fusion.fusion_candidates.preliminary

Common issues
=============

1.  Make sure to set the 'chimSegmentMin' option in the STAR tool to a desired minimum chimeric segment length in order to get the proper input files for the STAR Fusion tool.

Contatiner
==========

Docker image auto-builds at `cgrlab/star-fusion`

STAR-Fusion 0.8.0 <https://hub.docker.com/r/cgrlab/star-fusion/>

I/O
===

Inputs
------

Outputs
-------

Arguments
---------
